src/Grapevine/Client/RestClient.cs

```diff
@@ -66,8 +66,11 @@ public class RestClient : IRestClient
         public CookieContainer Cookies { get; }
         public ICredentials Credentials { get; set; }
 
-        public RestClient()
+        private readonly Action _timeoutDelegate;
+
+        public RestClient(Action timeoutDelegate = null)
         {
+            _timeoutDelegate = timeoutDelegate;
```

Comment:
`RestClient` has to have a parameterless constructor. How about moving this to a public property?